diff -ur o-railties-0.10.1/bin/breakpointer railties-0.10.1/bin/breakpointer
--- o-railties-0.10.1/bin/breakpointer	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/bin/breakpointer	2005-03-19 22:27:29.000000000 -0700
@@ -1,3 +1,2 @@
-#!/usr/local/bin/ruby
-$LOAD_PATH << File.dirname(__FILE__) + '/../vendor/railties/lib'
-require 'breakpoint_client'
+#!/usr/bin/ruby
+require 'railties/breakpoint_client'
diff -ur o-railties-0.10.1/bin/console railties-0.10.1/bin/console
--- o-railties-0.10.1/bin/console	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/bin/console	2005-03-19 23:20:18.000000000 -0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 irb = RUBY_PLATFORM =~ /mswin32/ ? 'irb.bat' : 'irb'
 
 require 'optparse'
diff -ur o-railties-0.10.1/bin/destroy railties-0.10.1/bin/destroy
--- o-railties-0.10.1/bin/destroy	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/bin/destroy	2005-03-19 23:33:47.000000000 -0700
@@ -1,7 +1,7 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 require File.dirname(__FILE__) + '/../config/environment'
-require 'rails_generator'
-require 'rails_generator/scripts/destroy'
+require 'railties/rails_generator'
+require 'railties/rails_generator/scripts/destroy'
 
 ARGV.shift if ['--help', '-h'].include?(ARGV[0])
 Rails::Generator::Scripts::Destroy.new.run(ARGV)
diff -ur o-railties-0.10.1/bin/generate railties-0.10.1/bin/generate
--- o-railties-0.10.1/bin/generate	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/bin/generate	2005-03-19 23:22:03.963568650 -0700
@@ -1,7 +1,7 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 require File.dirname(__FILE__) + '/../config/environment'
-require 'rails_generator'
-require 'rails_generator/scripts/generate'
+require 'railties/rails_generator'
+require 'railties/rails_generator/scripts/generate'
 
 ARGV.shift if ['--help', '-h'].include?(ARGV[0])
 Rails::Generator::Scripts::Generate.new.run(ARGV)
diff -ur o-railties-0.10.1/bin/rails railties-0.10.1/bin/rails
--- o-railties-0.10.1/bin/rails	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/bin/rails	2005-03-19 23:34:30.000000000 -0700
@@ -1,6 +1,8 @@
+#!/usr/bin/ruby
+
 abort "Rails requires Ruby 1.8.2" if RUBY_VERSION < "1.8.2"
 
-require File.dirname(__FILE__) + '/../lib/rails_generator'
-require 'rails_generator/scripts/generate'
+require 'railties/rails_generator'
+require 'railties/rails_generator/scripts/generate'
 Rails::Generator::Base.use_application_sources!
-Rails::Generator::Scripts::Generate.new.run(ARGV, :generator => 'app')
\ No newline at end of file
+Rails::Generator::Scripts::Generate.new.run(ARGV, :generator => 'app')
diff -ur o-railties-0.10.1/bin/server railties-0.10.1/bin/server
--- o-railties-0.10.1/bin/server	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/bin/server	2005-03-19 23:21:27.369041446 -0700
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 
 require 'webrick'
 require 'optparse'
@@ -40,7 +40,7 @@
 
 ENV["RAILS_ENV"] = OPTIONS[:environment]
 require File.dirname(__FILE__) + "/../config/environment"
-require 'webrick_server'
+require 'railties/webrick_server'
 
 OPTIONS['working_directory'] = File.expand_path(RAILS_ROOT)
 
diff -ur o-railties-0.10.1/bin/update railties-0.10.1/bin/update
--- o-railties-0.10.1/bin/update	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/bin/update	2005-03-19 23:35:31.000000000 -0700
@@ -1,5 +1,5 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 require File.dirname(__FILE__) + '/../config/environment'
-require 'rails_generator'
-require 'rails_generator/scripts/update'
+require 'railties/rails_generator'
+require 'railties/rails_generator/scripts/update'
 Rails::Generator::Scripts::Update.new.run(ARGV)
diff -ur o-railties-0.10.1/dispatches/dispatch.fcgi railties-0.10.1/dispatches/dispatch.fcgi
--- o-railties-0.10.1/dispatches/dispatch.fcgi	2005-03-19 22:20:06.000000000 -0700
+++ railties-0.10.1/dispatches/dispatch.fcgi	2005-03-19 23:18:56.000000000 -0700
@@ -1,7 +1,7 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 
 require File.dirname(__FILE__) + "/../config/environment"
-require 'dispatcher'
+require 'railties/dispatcher'
 require 'fcgi'
 
 log_file_path = "#{RAILS_ROOT}/log/fastcgi.crash.log"
@@ -17,4 +17,4 @@
       STDERR << "Couldn't write to #{log_file_path} (#{log_error} [#{log_error.class}])\n" << error_message
     end
   end
-end
\ No newline at end of file
+end
diff -ur o-railties-0.10.1/dispatches/dispatch.rb railties-0.10.1/dispatches/dispatch.rb
--- o-railties-0.10.1/dispatches/dispatch.rb	2005-03-19 22:20:06.000000000 -0700
+++ railties-0.10.1/dispatches/dispatch.rb	2005-03-19 22:27:29.000000000 -0700
@@ -1,10 +1,8 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 
 require File.dirname(__FILE__) + "/../config/environment" unless defined?(RAILS_ROOT)
 
-# If you're using RubyGems and mod_ruby, this require should be changed to an absolute path one, like:
-# "/usr/local/lib/ruby/gems/1.8/gems/rails-0.8.0/lib/dispatcher" -- otherwise performance is severely impaired
-require "dispatcher"
+require "railties/dispatcher"
 
 ADDITIONAL_LOAD_PATHS.reverse.each { |dir| $:.unshift(dir) if File.directory?(dir) } if defined?(Apache::RubyRun)
-Dispatcher.dispatch
\ No newline at end of file
+Dispatcher.dispatch
diff -ur o-railties-0.10.1/environments/shared.rb railties-0.10.1/environments/shared.rb
--- o-railties-0.10.1/environments/shared.rb	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/environments/shared.rb	2005-03-19 23:19:27.000000000 -0700
@@ -19,14 +19,6 @@
   config
   components
   lib
-  vendor
-  vendor/railties
-  vendor/railties/lib
-  vendor/activesupport/lib
-  vendor/activerecord/lib
-  vendor/actionpack/lib
-  vendor/actionmailer/lib
-  vendor/actionwebservice/lib
 ).map { |dir| "#{RAILS_ROOT}/#{dir}" }
 
 # Prepend to $LOAD_PATH
diff -ur o-railties-0.10.1/fresh_rakefile railties-0.10.1/fresh_rakefile
--- o-railties-0.10.1/fresh_rakefile	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/fresh_rakefile	2005-03-19 23:22:42.165898446 -0700
@@ -75,29 +75,11 @@
   rdoc.options << '--line-numbers --inline-source'
   rdoc.rdoc_files.include('README')
   rdoc.rdoc_files.include('CHANGELOG')
-  rdoc.rdoc_files.include('vendor/railties/CHANGELOG')
-  rdoc.rdoc_files.include('vendor/railties/MIT-LICENSE')
-  rdoc.rdoc_files.include('vendor/activerecord/README')
-  rdoc.rdoc_files.include('vendor/activerecord/CHANGELOG')
-  rdoc.rdoc_files.include('vendor/activerecord/lib/active_record/**/*.rb')
-  rdoc.rdoc_files.exclude('vendor/activerecord/lib/active_record/vendor/*')
-  rdoc.rdoc_files.include('vendor/actionpack/README')
-  rdoc.rdoc_files.include('vendor/actionpack/CHANGELOG')
-  rdoc.rdoc_files.include('vendor/actionpack/lib/action_controller/**/*.rb')
-  rdoc.rdoc_files.include('vendor/actionpack/lib/action_view/**/*.rb')
-  rdoc.rdoc_files.include('vendor/actionmailer/README')
-  rdoc.rdoc_files.include('vendor/actionmailer/CHANGELOG')
-  rdoc.rdoc_files.include('vendor/actionmailer/lib/action_mailer/base.rb')
-  rdoc.rdoc_files.include('vendor/actionwebservice/README')
-  rdoc.rdoc_files.include('vendor/actionwebservice/ChangeLog')
-  rdoc.rdoc_files.include('vendor/actionwebservice/lib/action_web_service/**/*.rb')
-  rdoc.rdoc_files.include('vendor/activesupport/README')
-  rdoc.rdoc_files.include('vendor/activesupport/lib/active_support/**/*.rb')
 }
 
 desc "Report code statistics (KLOCs, etc) from the application"
 task :stats => [ :environment ] do
-  require 'code_statistics'
+  require 'railties/code_statistics'
   CodeStatistics.new(
     ["Helpers", "app/helpers"], 
     ["Controllers", "app/controllers"], 
diff -ur o-railties-0.10.1/lib/breakpoint.rb railties-0.10.1/lib/breakpoint.rb
--- o-railties-0.10.1/lib/breakpoint.rb	2005-03-19 22:20:06.000000000 -0700
+++ railties-0.10.1/lib/breakpoint.rb	2005-03-19 22:27:40.000000000 -0700
@@ -16,7 +16,7 @@
 # license please contact me.
 
 require 'irb'
-require 'binding_of_caller'
+require 'railties/binding_of_caller'
 require 'drb'
 require 'drb/acl'
 
diff -ur o-railties-0.10.1/lib/dispatcher.rb railties-0.10.1/lib/dispatcher.rb
--- o-railties-0.10.1/lib/dispatcher.rb	2005-03-19 22:20:06.000000000 -0700
+++ railties-0.10.1/lib/dispatcher.rb	2005-03-19 22:27:40.000000000 -0700
@@ -21,7 +21,7 @@
 # WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #++
 
-require 'breakpoint'
+require 'railties/breakpoint'
 
 class Dispatcher
   class << self
diff -ur o-railties-0.10.1/lib/rails_generator.rb railties-0.10.1/lib/rails_generator.rb
--- o-railties-0.10.1/lib/rails_generator.rb	2005-03-19 22:20:06.000000000 -0700
+++ railties-0.10.1/lib/rails_generator.rb	2005-03-19 23:36:44.975381074 -0700
@@ -22,22 +22,16 @@
 #++
 
 $:.unshift(File.dirname(__FILE__))
-$:.unshift(File.dirname(__FILE__) + "/../../activesupport/lib")
 
-begin
-  require 'active_support'  
-rescue LoadError
-  require 'rubygems'
-  require_gem 'activesupport'
-end
+require 'active_support'  
 
-require 'rails_generator/base'
-require 'rails_generator/lookup'
-require 'rails_generator/commands'
+require 'railties/rails_generator/base'
+require 'railties/rails_generator/lookup'
+require 'railties/rails_generator/commands'
 
 Rails::Generator::Base.send(:include, Rails::Generator::Lookup)
 Rails::Generator::Base.send(:include, Rails::Generator::Commands)
 
 # Set up a default logger for convenience.
-require 'rails_generator/simple_logger'
+require 'railties/rails_generator/simple_logger'
 Rails::Generator::Base.logger = Rails::Generator::SimpleLogger.new(STDOUT)
diff -ur o-railties-0.10.1/Rakefile railties-0.10.1/Rakefile
--- o-railties-0.10.1/Rakefile	2005-03-19 22:20:07.000000000 -0700
+++ railties-0.10.1/Rakefile	2005-03-19 22:30:52.000000000 -0700
@@ -1,8 +1,6 @@
 require 'rake'
 require 'rake/testtask'
 require 'rake/rdoctask'
-require 'rake/gempackagetask'
-require 'rake/contrib/rubyforgepublisher'
 
 require 'date'
 require 'rbconfig'
@@ -14,7 +12,7 @@
 PKG_DESTINATION = ENV["RAILS_PKG_DESTINATION"] || "../#{PKG_NAME}"
 
 
-BASE_DIRS   = %w( app config/environments components db doc log lib public script test vendor )
+BASE_DIRS   = %w( app config/environments components db doc log lib public script test )
 APP_DIRS    = %w( apis models controllers helpers views views/layouts )
 PUBLIC_DIRS = %w( images javascripts stylesheets )
 TEST_DIRS   = %w( fixtures unit functional mocks mocks/development mocks/test )
@@ -23,26 +21,23 @@
 HTML_FILES   = %w( 404.html 500.html index.html favicon.ico )
 BIN_FILES    = %w( generate destroy breakpointer console server update )
 
-VENDOR_LIBS = %w( actionpack activerecord actionmailer activesupport actionwebservice railties )
-
-
 desc "Default Task"
 task :default => [ :fresh_rails ]
 
 desc "Generates a fresh Rails package with documentation"
-task :fresh_rails => [ :clean, :make_dir_structure, :initialize_file_stubs, :copy_vendor_libraries, :copy_ties_content, :generate_documentation ]
+task :fresh_rails => [ :clean, :make_dir_structure, :initialize_file_stubs, :copy_ties_content, :generate_documentation ]
 
 desc "Generates a fresh Rails package using GEMs with documentation"
 task :fresh_gem_rails => [ :clean, :make_dir_structure, :initialize_file_stubs, :copy_ties_content, :copy_gem_environment ]
 
 desc "Generates a fresh Rails package without documentation (faster)"
-task :fresh_rails_without_docs => [ :clean, :make_dir_structure, :initialize_file_stubs, :copy_vendor_libraries, :copy_ties_content ]
+task :fresh_rails_without_docs => [ :clean, :make_dir_structure, :initialize_file_stubs, :copy_ties_content ]
 
 desc "Generates a fresh Rails package without documentation (faster)"
-task :fresh_rails_without_docs_using_links => [ :clean, :make_dir_structure, :initialize_file_stubs, :link_vendor_libraries, :copy_ties_content ]
+task :fresh_rails_without_docs_using_links => [ :clean, :make_dir_structure, :initialize_file_stubs, :copy_ties_content ]
 
 desc "Generates minimal Rails package using symlinks"
-task :dev => [ :clean, :make_dir_structure, :initialize_file_stubs, :link_vendor_libraries, :copy_ties_content ]
+task :dev => [ :clean, :make_dir_structure, :initialize_file_stubs, :copy_ties_content ]
 
 desc "Packages the fresh Rails package with documentation"
 task :package => [ :clean, :fresh_rails ] do
@@ -85,24 +80,6 @@
   end
 end
 
-
-# Copy Vendors ----------------------------------------------------------------------------
-
-desc "Copy in all the Rails packages to vendor"
-task :copy_vendor_libraries do
-  cp_r VENDOR_LIBS.map { |dir| File.join('..', dir) },
-       File.join(PKG_DESTINATION, 'vendor')
-end
-
-desc "Link in all the Rails packages to vendor"
-task :link_vendor_libraries do
-  return_dir = File.dirname(File.expand_path(__FILE__))
-  cd File.join(PKG_DESTINATION, 'vendor')
-  VENDOR_LIBS.each { |dir| ln_s File.dirname(__FILE__) + "/../../#{dir}", "." }
-  cd return_dir
-end
-
-
 # Copy Ties Content -----------------------------------------------------------------------
 
 # :link_apache_config
@@ -267,4 +244,4 @@
 task :pgem => [:gem] do 
   Rake::SshFilePublisher.new("davidhh@comox.textdrive.com", "public_html/gems/gems", "pkg", "#{PKG_FILE_NAME}.gem").upload
   `ssh davidhh@comox.textdrive.com './gemupdate.sh'`
-end
\ No newline at end of file
+end
