diff -ur o-rails/vendor/rails/railties/bin/breakpointer rails/vendor/rails/railties/bin/breakpointer
--- o-rails/vendor/rails/railties/bin/breakpointer	2005-03-27 08:01:37.000000000 -0700
+++ rails/vendor/rails/railties/bin/breakpointer	2005-04-03 14:00:57.022096171 -0600
@@ -1,3 +1,2 @@
-#!/usr/local/bin/ruby
-$LOAD_PATH << File.dirname(__FILE__) + '/../vendor/railties/lib'
-require 'breakpoint_client'
+#!/usr/bin/ruby
+require 'railties/breakpoint_client'
diff -ur o-rails/vendor/rails/railties/bin/console rails/vendor/rails/railties/bin/console
--- o-rails/vendor/rails/railties/bin/console	2005-03-27 08:01:37.000000000 -0700
+++ rails/vendor/rails/railties/bin/console	2005-04-03 14:00:57.023096047 -0600
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 irb = RUBY_PLATFORM =~ /mswin32/ ? 'irb.bat' : 'irb'
 
 require 'optparse'
diff -ur o-rails/vendor/rails/railties/bin/destroy rails/vendor/rails/railties/bin/destroy
--- o-rails/vendor/rails/railties/bin/destroy	2005-03-27 08:01:37.000000000 -0700
+++ rails/vendor/rails/railties/bin/destroy	2005-04-03 14:00:57.024095923 -0600
@@ -1,7 +1,7 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 require File.dirname(__FILE__) + '/../config/environment'
-require 'rails_generator'
-require 'rails_generator/scripts/destroy'
+require 'railties/rails_generator'
+require 'railties/rails_generator/scripts/destroy'
 
 ARGV.shift if ['--help', '-h'].include?(ARGV[0])
 Rails::Generator::Scripts::Destroy.new.run(ARGV)
diff -ur o-rails/vendor/rails/railties/bin/generate rails/vendor/rails/railties/bin/generate
--- o-rails/vendor/rails/railties/bin/generate	2005-03-27 08:01:37.000000000 -0700
+++ rails/vendor/rails/railties/bin/generate	2005-04-03 14:00:57.025095799 -0600
@@ -1,7 +1,7 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 require File.dirname(__FILE__) + '/../config/environment'
-require 'rails_generator'
-require 'rails_generator/scripts/generate'
+require 'railties/rails_generator'
+require 'railties/rails_generator/scripts/generate'
 
 ARGV.shift if ['--help', '-h'].include?(ARGV[0])
 Rails::Generator::Scripts::Generate.new.run(ARGV)
diff -ur o-rails/vendor/rails/railties/bin/rails rails/vendor/rails/railties/bin/rails
--- o-rails/vendor/rails/railties/bin/rails	2005-03-27 08:01:37.000000000 -0700
+++ rails/vendor/rails/railties/bin/rails	2005-04-03 14:00:57.026095676 -0600
@@ -1,6 +1,8 @@
+#!/usr/bin/ruby
+
 abort "Rails requires Ruby 1.8.2" if RUBY_VERSION < "1.8.2"
 
-require File.dirname(__FILE__) + '/../lib/rails_generator'
-require 'rails_generator/scripts/generate'
+require 'railties/rails_generator'
+require 'railties/rails_generator/scripts/generate'
 Rails::Generator::Base.use_application_sources!
-Rails::Generator::Scripts::Generate.new.run(ARGV, :generator => 'app')
\ No newline at end of file
+Rails::Generator::Scripts::Generate.new.run(ARGV, :generator => 'app')
diff -ur o-rails/vendor/rails/railties/bin/server rails/vendor/rails/railties/bin/server
--- o-rails/vendor/rails/railties/bin/server	2005-03-27 08:01:37.000000000 -0700
+++ rails/vendor/rails/railties/bin/server	2005-04-03 14:00:57.027095552 -0600
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 
 require 'webrick'
 require 'optparse'
@@ -40,7 +40,7 @@
 
 ENV["RAILS_ENV"] = OPTIONS[:environment]
 require File.dirname(__FILE__) + "/../config/environment"
-require 'webrick_server'
+require 'railties/webrick_server'
 
 OPTIONS['working_directory'] = File.expand_path(RAILS_ROOT)
 
diff -ur o-rails/vendor/rails/railties/bin/update rails/vendor/rails/railties/bin/update
--- o-rails/vendor/rails/railties/bin/update	2005-03-27 08:01:37.000000000 -0700
+++ rails/vendor/rails/railties/bin/update	2005-04-03 14:00:57.028095428 -0600
@@ -1,5 +1,5 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 require File.dirname(__FILE__) + '/../config/environment'
-require 'rails_generator'
-require 'rails_generator/scripts/update'
+require 'railties/rails_generator'
+require 'railties/rails_generator/scripts/update'
 Rails::Generator::Scripts::Update.new.run(ARGV)
diff -ur o-rails/vendor/rails/railties/dispatches/dispatch.fcgi rails/vendor/rails/railties/dispatches/dispatch.fcgi
--- o-rails/vendor/rails/railties/dispatches/dispatch.fcgi	2005-03-27 08:01:37.000000000 -0700
+++ rails/vendor/rails/railties/dispatches/dispatch.fcgi	2005-04-03 14:04:28.804886433 -0600
@@ -1,4 +1,4 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 
 def dispatcher_error(path,e,msg="")
   error_message =
@@ -10,7 +10,7 @@
 
 begin
   require File.dirname(__FILE__) + "/../config/environment"
-  require 'dispatcher'
+  require 'railties/dispatcher'
   require 'fcgi'
 
   log_file_path = "#{RAILS_ROOT}/log/fastcgi.crash.log"
@@ -24,4 +24,4 @@
   end
 rescue Object => fcgi_error
   dispatcher_error(log_file_path, fcgi_error, "FCGI process #{$$} killed by this error\n")
-end
\ No newline at end of file
+end
Only in rails/vendor/rails/railties/dispatches: dispatch.fcgi.orig
diff -ur o-rails/vendor/rails/railties/dispatches/dispatch.rb rails/vendor/rails/railties/dispatches/dispatch.rb
--- o-rails/vendor/rails/railties/dispatches/dispatch.rb	2005-03-27 08:01:37.000000000 -0700
+++ rails/vendor/rails/railties/dispatches/dispatch.rb	2005-04-03 14:00:57.034094686 -0600
@@ -1,10 +1,8 @@
-#!/usr/local/bin/ruby
+#!/usr/bin/ruby
 
 require File.dirname(__FILE__) + "/../config/environment" unless defined?(RAILS_ROOT)
 
-# If you're using RubyGems and mod_ruby, this require should be changed to an absolute path one, like:
-# "/usr/local/lib/ruby/gems/1.8/gems/rails-0.8.0/lib/dispatcher" -- otherwise performance is severely impaired
-require "dispatcher"
+require "railties/dispatcher"
 
 ADDITIONAL_LOAD_PATHS.reverse.each { |dir| $:.unshift(dir) if File.directory?(dir) } if defined?(Apache::RubyRun)
-Dispatcher.dispatch
\ No newline at end of file
+Dispatcher.dispatch
